scripts/lib/CIME/XML/component.py

```diff
@@ -282,3 +282,46 @@ def print_values(self):
             name = self.get(entry, "id")
             text = self.text(self.get_child("desc", root=entry))
             logger.info("   {:20s} : {}".format(name, text.encode('utf-8')))
+
+
+    def return_values(self):
+        """
+        return a list of hashes from target config_component.xml file
+        """
+        entry_dict = dict()
+        items = list()
+        try:
+            helpnode = self.get_child("help")
+            helptext = self.text(helpnode)
+        except:
```

Comment:
I think that you need to specify an exception here, py3 does not allow general exceptions.

Reply:
What would be the exception returned if the self.text(helpnode) does not exist?